### PR TITLE
[WIP] pg-mvt: Remove dependency on cartodb-postgresql extension

### PR DESCRIPTION
**Pull request by Algunenano**

### CODE REVIEW

**Comment by Algunenano** [comment]
I'm reviewing Windshaft-cartodb with these changes, since the results of the aggregations depend on rounding :cry:

File: lib/windshaft/renderers/pg-mvt/renderer.js [hunk 1]
```diff
@@ -268,7 +271,7 @@ module.exports = class PostgresVectorRenderer {
 
                 const stats = timer.getTimes();
 
-                return resolve([mvt || new Buffer(0), headers, stats]);
+                return resolve([mvt || new Buffer.alloc(0), headers, stats]);
```
Comment (dgaubert) [inline lib/windshaft/renderers/pg-mvt/renderer.js]: ❤️

File: lib/windshaft/utils/cartodb_postgis_utils.js [hunk 2]
```diff
@@ -8,22 +8,19 @@ function CartoDBPostgisUtils (tileSize, maxGeosize) {
 module.exports = CartoDBPostgisUtils;
 
 CartoDBPostgisUtils.prototype.cdbXYZResolution = function (z) {
-    var full_resolution = this.tileMaxGeosize / this.tileSize;
+    const full_resolution = this.tileMaxGeosize / this.tileSize;
     return full_resolution / Math.pow(2, z);
 };
 
 CartoDBPostgisUtils.prototype.cdbXYZExtent = function (x, y, z) {
```
Comment (dgaubert) [inline lib/windshaft/utils/cartodb_postgis_utils.js]: Having in mind this module was only used by the old and deprecated `GeoJSON` module, we can modify it to our current needs.

**Comment by dgaubert** [comment]
In `windshaft-cartodb` we still rely on CartoDB extension by using functions such as `CDB_XYZ_Resolution` and  `CDB_ZoomFromScale`.

**Comment by dgaubert** [comment]
^^ here: https://github.com/CartoDB/Windshaft-cartodb/blob/master/lib/cartodb/models/aggregation/aggregation-query.js#L304

File: lib/windshaft/utils/cartodb_postgis_utils.js [hunk 3]
```diff
@@ -1,29 +1,50 @@
...
 
+const Decimal = require('decimal.js');
+
 function CartoDBPostgisUtils (tileSize, maxGeosize) {
     this.tileSize = tileSize || 256;
-    this.tileMaxGeosize = maxGeosize || 40075017; // earth circumference in webmercator 3857
+    if (maxGeosize) {
+        this.tileMaxGeosize = new Decimal(maxGeosize);
+    } else {
+        // 6378137: earth circumference in webmercator 3857
```
Comment (rafatower) [inline lib/windshaft/utils/cartodb_postgis_utils.js]: not to be picky, but isn't it the radius?

Comment (Algunenano) [reply lib/windshaft/utils/cartodb_postgis_utils.js]: Yup, totally right

File: lib/windshaft/index.js [hunk 4]
```diff
@@ -25,6 +25,7 @@ module.exports = {
     },
     backend: require('./backends'),
     version: version,
+    cartodb_utils: require('./utils/cartodb_postgis_utils'),
```
Comment (dgaubert) [inline lib/windshaft/index.js]: Let's try to use `CamelCase` at least for the public interface

Comment (dgaubert) [reply lib/windshaft/index.js]: what about naming it `utils`. Then in the client:

```js
const windshaft = require('windhsaft');
windshaft.utils.someMethod();
```

Comment (dgaubert) [reply lib/windshaft/index.js]: Nevermind, it's a class object. :(

Comment (dgaubert) [reply lib/windshaft/index.js]: Just thinking about the public interface:

```js
const windshaft = require('windhsaft');
const { TileHelper } = windshaft;
const tileSize = 256;
const tileHelper = new TileHelper(tileSize);
const resolution = tileHelper.getResolution({ z: 0 });
const extent = tileHelper.getExtent({ z: 0, x: 0, y: 0 });
```

thoughts?

Comment (Algunenano) [reply lib/windshaft/index.js]: The idea was to keep the same names / parameters to match the `cartodb` extension functions, but maybe changing to webMercatorUtils and providing getResoluton and getExtent as you say is clearer.

Comment (dgaubert) [reply lib/windshaft/index.js]: IIRC, we didn't use that module so we can rename it if we want to. `WebMercatorHelper` sounds great, no?

**Pull request merged**